### DISCUSSION TITLE
How to detect when a transaction is speed up

### DISCUSSION THREAD

**Discussion by davigp** [wevm/wagmi#799] [Q&A]
While using `useWaitForTransaction` after the transaction is sped up, it keeps waiting forever. Is EIP-2831 supported, or any other way to retrieve the new hash when the original transaction is replaced on a speed-up request?

References: 
- https://github.com/ethereum/EIPs/pull/2831/
- https://github.com/MetaMask/metamask-extension/issues/7227
- https://github.com/ChainSafe/web3.js/issues/3723

**Comment by tmm** [comment]
Currently not supported. You could try listening for new messages on the active connector's (`const { connector } = useAccount()`) provider (`connector.getProvider()`).

**Reply by davigp** [reply]
Thanks @tmm , I'll try it and let you know how it goes

**Reply by sarinniural** [reply]
@davigp were you able to come up with a solution for this?

**Comment by StKostyantin** [comment]
@tmm when will work on this start?

**Reply by tmm** [reply]
No ETA. Working on other things at the moment. If/when work on this starts, I will update this discussion.

**Reply by StKostyantin** [reply]
In fact, we already have everything to at least throw an exception.

In `useWaitForTransaction`, we pass a `hash` that we take from `useContractWrite`

in a couple of minutes I created such a hook (which can be improved), but its logic can be inserted into useWaitForTransaction to avoid infinite loading

```
export const useCheckSpeedUpTrx = ({ hash }: { hash?: `0x${string}` }) => {
  const [isSpeedUp, setIsSpeedUp] = useState(false);
  const [isStop, setIsStop] = useState(false);
  const provider = useProvider();

  const handleGet = async (hash: `0x${string}`) => {
    try {
      const res = await provider.getTransaction(hash);
      if (res.confirmations) {
        setIsStop(true);
      }
    } catch (e) {
      setIsSpeedUp(true);
    }
  };

  useEffect(() => {
    if (hash) {
      const interval = setInterval(() => handleGet(hash), 5000);
      (isSpeedUp || isStop) && clearInterval(interval);
      return () => clearInterval(interval);
    }
  }, [hash, isSpeedUp, isStop]);

  return { isSpeedUp };
};
```

**Comment by tmm** [comment]
This will be released in `wagmi@0.9` (See https://github.com/wagmi-dev/wagmi/pull/1344 for more info)

**Comment by emmajane1313** [comment]
@tmm was it released? Onspeedup doesnt seem to work still, on speeding up the transaction the function inside onspeedup for waitfortransaction doesnt fire.
